src/instrumentation/opentelemetry.ts

```diff
@@ -116,3 +127,26 @@ export function buildTraceTree({
     },
   );
 }
+
+export const isMeasurementedInstance = (instance: any) =>
+  Boolean(
+    instance &&
+      (instance instanceof BaseAgent || instance instanceof Tool || instance instanceof BaseLLM),
+  );
+
+export function buildModuleUsageMetric({
+  instance,
+  traceId,
+  eventId,
+}: BuildModuleUsageMetricProps) {
+  moduleUsageGauge.record(1, {
+    source: instance.constructor.name,
+    type: instance instanceof BaseAgent ? "agent" : instance instanceof Tool ? "tool" : "llm",
+    framework_version: Version,
+    os_type: os.type(),
+    os_release: os.release(),
+    os_arch: os.arch(),
+    trace_id: traceId,
+    event_id: eventId,
+  });
```

Comment:
It is weird to see most of these labels,

os, fw version, trace are likely supplied by the SDK (not necessarily as a label but as OTEL resource or something like that)

trace_id and event_id are not a good fit for a metric

source and type are good but it signals this should be an incrementing counter not a gauge

Lastly, and this is likely the most important. I'd say we'd want a label that establishes an anonymous identity once framework is cloned (or first used). A random ID that's stable.